tuf/repository_tool.py

```diff
@@ -2506,84 +2505,41 @@ def delegate_hashed_bins(self, list_of_targets, keys_of_hashed_bins,
     logger.info(repr(len(list_of_targets)) + ' total targets.')
     logger.info(repr(number_of_bins) + ' hashed bins.')
     logger.info(repr(total_hash_prefixes) + ' total hash prefixes.')
+    logger.info('Each bin ranges over ' + repr(bin_size) + ' hash prefixes.')
+
+    # Generate a list of bin names, the range of prefixes to be delegated to
+    # that bin, along with the corresponding full list of target prefixes
+    # to be delegated to that bin
+    ordered_roles = []
+    for idx in range(0, total_hash_prefixes, bin_size):
+      high = idx + bin_size - 1
+      name = _create_bin_name(idx, high, prefix_length)
+      if bin_size == 1:
+        target_hash_prefixes = [name]
+      else:
+        target_hash_prefixes = []
+        for idy in range(idx, idx+bin_size):
+          target_hash_prefixes.append("{prefix:0{len}x}".format(prefix=idy,
+              len=prefix_length))
 
-    # Store the target paths that fall into each bin.  The digest of the target
-    # path, reduced to the first 'prefix_length' hex digits, is calculated to
-    # determine which 'bin_index' it should go.  we use xrange() here because
-    # there can be a large number of prefixes to process.
-    target_paths_in_bin = {}
-    for bin_index in six.moves.xrange(total_hash_prefixes):
-      target_paths_in_bin[bin_index] = []
+      role = {"name": name, "target_paths": [],
+          'target_hash_prefixes': target_hash_prefixes}
```

Comment:
Minor nit: I think dict literals are easier to read when each item is on a separate line.

Reply:
I've squashed in a change to make each item on a separate line _and_ use consistent quotation marks.